tutorials/src/bin/basic-tutorial-5.rs

```diff
@@ -0,0 +1,377 @@
+    }
+
+    add_streams_info(playbin, textbufcell, "video".to_string());
+    add_streams_info(playbin, textbufcell, "audio".to_string());
+    add_streams_info(playbin, textbufcell, "text".to_string());
+}
+
+#[cfg(feature = "tutorial5")]
+// This creates all the GTK+ widgets that compose our application, and registers the callbacks
+fn create_ui(playbin: &gst::Element) {
+    let main_window = Window::new(WindowType::Toplevel);
+    main_window.connect_delete_event(|_, _| {
+        gtk::main_quit();
+        Inhibit(false)
+    });
+
+    let play_button =
+        gtk::Button::new_from_icon_name("media-playback-start", gtk::IconSize::SmallToolbar.into());
+    let pipeline = playbin.clone();
+    play_button.connect_clicked(move |_| {
+        let pipeline = &pipeline;
+        pipeline.set_state(gst::State::Playing);
+    });
+
+    let pause_button =
+        gtk::Button::new_from_icon_name("media-playback-pause", gtk::IconSize::SmallToolbar.into());
+    let pipeline = playbin.clone();
+    pause_button.connect_clicked(move |_| {
+        let pipeline = &pipeline;
+        pipeline.set_state(gst::State::Paused);
+    });
+
+    let stop_button =
+        gtk::Button::new_from_icon_name("media-playback-stop", gtk::IconSize::SmallToolbar.into());
+    let pipeline = playbin.clone();
+    stop_button.connect_clicked(move |_| {
+        let pipeline = &pipeline;
+        pipeline.set_state(gst::State::Ready);
+    });
+
+    let slider = gtk::Scale::new_with_range(
+        gtk::Orientation::Horizontal,
+        0.0 as f64,
+        100.0 as f64,
+        1.0 as f64,
+    );
+    let pipeline = playbin.clone();
+    let slider_update_signal_id = slider
+        .connect_value_changed(move |slider| {
+            let pipeline = &pipeline;
+            let value = slider.get_value() as u64;
+            if pipeline
+                .seek_simple(
+                    gst::Format::Time,
+                    gst::SeekFlags::FLUSH | gst::SeekFlags::KEY_UNIT,
+                    ((value * gst::SECOND) as i64),
+                )
+                .is_err()
+            {
+                eprintln!("Seeking to {} failed", value);
+            }
+        })
+        .to_glib();
+    slider.set_draw_value(false);
+    let pipeline = playbin.clone();
+    let lslider = slider.clone();
+    // Update the UI (seekbar) every second
+    gtk::timeout_add_seconds(1, move || {
+        let pipeline = &pipeline;
+        let lslider = &lslider;
+
+        match pipeline.query_duration(gst::Format::Time) {
+            Some(x) => {
+                let seconds = (x as u64) / gst::SECOND;
+                lslider.set_range(0.0, seconds as f64);
+            }
+            None => (),
+        }
+
+        let position = pipeline.query_position(gst::Format::Time);
+        if let Some(position) = position {
+            let seconds = (position as u64) / gst::SECOND;
+            signal::signal_handler_block(lslider, from_glib(slider_update_signal_id));
```

Comment:
The `from_glib` should not be needed at all here. If `connect_value_changed` returns an `u64` instead, you found a bug in gtk-rs and can submit a PR there :)

Reply:
It returns a `SignalHandlerId` no a u64 as all other signals iiuc.

Reply:
To be fixed then :)